src/states/elder/mod.rs

```diff
@@ -1810,6 +1853,22 @@ impl Approved for Elder {
         }
         Ok(())
     }
+
+    fn before_handle_parsec_gossip(&mut self, parsec_version: u64, sender: &PublicId) {
+        if parsec_version >= self.parsec_map.last_version() {
+            // This member has caught up to the latest parsec version. We don't need to track its
+            // knowledge any more.
+            let _ = self.members_knowledge.remove(sender.name());
```

Comment:
Don't we still need to track it, what about when it is demoted?

Reply:
We re-insert it when the member is demoted.

Reply:
(ok, though probably with a version the node has not reached yet, but that is part of other comment thread).